### ISSUE TITLE
Annoying venv warning

### ISSUE THREAD

**Issue by jamilraichouni** [ipython/ipython#13875]
Hi!

Using ipython 8.7.0 I just identified that the block below always triggers the annoying warning when I run ipython within a pyenv venv.

https://github.com/ipython/ipython/blob/d38397b078b744839b8510f7ac9ab4fa40450a4f/IPython/core/interactiveshell.py#L886-L890

The instance variable `warn_venv` occurs two times in the whole module (here both locations in the snippets cutout) and seems to be always true.

https://github.com/ipython/ipython/blob/d38397b078b744839b8510f7ac9ab4fa40450a4f/IPython/core/interactiveshell.py#L511-L514

I see no point to always get that message although I clearly run the ipython executable in the currently active venv.

Maybe my symlink setup causes that?

Here you see somethin in my neovim session.
Top pane: the module
Middle pane: vars in a debug session in pdb
Bottom pane: my symlink of `$HOME/.pyenv` and the env var `VIRTUAL_ENV`

Maybe someone want to look deeper into that scenario. For now it looks like a warning that should not appear and I disable it.

I wish a Merry Christmas,
Jamil

<img width="1782" alt="image" src="https://user-images.githubusercontent.com/45113557/209349729-588e865b-f3be-472d-bea3-3e228d4b7a6d.png">

**Comment by cbrnr** [comment]
Is your IPython installed in the venv or do you use a global IPython? The latter will cause the warning to appear, and the solution is to disable it in your config.

**Comment by jamilraichouni** [comment]
The stuff runs in a Docker container. There I have a global Python 3.10.8 and a venv created via `pyenv`.
Both (globally and venv) have IPython installed.
I have the venv activated using a `.python-version` file and `oh-my-zsh` with the `python` plugin installed also indicates that in parallel to a set env var `VIRTUAL_ENV`.
When I then execute the command `ipython` in the root dir of my Python project/ repo I get that warning.

I can try to provide a `Dockerfile` that will hopefully give everything to be able to reproduce that scenario.
Would that help?

**Comment by cbrnr** [comment]
> Both (globally and venv) have IPython installed.

Maybe that's the problem. It might be that you start the global `ipython` even when your venv is activated. That's when you see the warning.

So either you start `ipython` installed in your venv, or you set `c.InteractiveShell.warn_venv = False` in your config.

Do these solutions not work for you? I found the warning also very annoying, and that's why I added the config option to suppress it.

**Comment by jamilraichouni** [comment]
Well the following screenshot tells (at least) me, that the IPython installed in the venv is being executed while I get the mentioned warning:

![image](https://user-images.githubusercontent.com/45113557/209399451-d0820463-73ed-4f8f-aa94-36d2708c1de1.png)

Thanks for the pointer to the cfg option. That is a better solution than commenting the warning. Nevertheless I assume that the warning is not supposed to appear anyway when I run the `ipython` executable that has been installed in the venv which is currently active.

**Comment by cbrnr** [comment]
Yes, it looks like you are running `ipython` from your env. To be sure, can you also activate your venv and then run `python -m IPython`? If the warning still appears, there might indeed be something wrong (maybe venv detection is not working in your case).

**Comment by jamilraichouni** [comment]
Doing so with the warning not disabled I get the same warning:

![image](https://user-images.githubusercontent.com/45113557/209401284-b820d6d8-9322-4e38-934b-b8225e227d84.png)
